pkg/sypgp/sypgp.go

```diff
@@ -628,79 +610,57 @@ func FetchPubkey(fingerprint, keyserverURI, authToken string, noPrompt bool) (op
 	if len(el) > 1 {
 		return nil, fmt.Errorf("server returned more than one key for unique fingerprint")
 	}
-
 	return el, nil
 }
 
-// doPushRequest prepares an HKP pks/add request
-func doPushRequest(w *bytes.Buffer, keyserverURI, authToken string) (*http.Request, error) {
-	v := url.Values{}
-	v.Set("keytext", w.String())
+func serializeEntity(e *openpgp.Entity, blockType string) (string, error) {
+	w := bytes.NewBuffer(nil)
 
-	u, err := url.Parse(keyserverURI)
+	wr, err := armor.Encode(w, blockType, nil)
 	if err != nil {
-		return nil, err
+		return "", err
 	}
-	u.Path = "pks/add"
 
-	r, err := http.NewRequest(http.MethodPost, u.String(), strings.NewReader(v.Encode()))
-	if err != nil {
-		return nil, err
-	}
-	if authToken != "" {
-		r.Header.Set("Authorization", fmt.Sprintf("BEARER %s", authToken))
+	if err = e.Serialize(wr); err != nil {
+		wr.Close()
```

Comment:
Possible to just do a `defer wr.Close()` before line 624 since we're calling `wr.Close()` before we return in either execution pathway?

Reply:
`wr` must be closed before `w.String()` will return the entire serialized entity below.